.github/workflows/publish.yml

```diff
@@ -13,53 +17,136 @@ on:
         required: false
         default: ''
         type: string
-      pure_python_wheel:
-        description: Whether to build a pure Python wheel - if not, separate wheels are built for each platform.
+      upload_to_pypi:
+        description: Always upload to PyPI - if not, only upload tags
         required: false
         default: false
         type: boolean
+      repository_url:
+        description: The PyPI repository URL to use
+        required: false
+        default: ''
+        type: string
+      submodules:
+        description: Whether to checkout submodules
+        required: false
+        default: true
+        type: boolean
+    secrets:
+      pypi_password:
+        required: false
 
 jobs:
 
+  targets:
+    name: Load build targets
+    runs-on: ubuntu-latest
+    outputs:
+      sdist: ${{ steps.set-outputs.outputs.sdist }}
+      universal: ${{ steps.set-outputs.outputs.universal }}
+      matrix: ${{ steps.set-outputs.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+# Uncomment before merging
+#        with:
+#           repository: 'OpenAstronomy/github-actions-workflows'
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      - run: python -m pip install PyYAML
+      - id: set-outputs
+        run: python tools/load_build_targets.py "${{ inputs.targets }}"
```

Comment:
This really is an amazing workaround! :clap: 

Dear Github actions: you are missing a critical feature.